### PR TITLE
AggregatedStatisticsActivity: Convert ListView to RecyclerView

### PR DESCRIPTION
**Pull request by lmarz**
Part of: #1613

Arguably, since there aren't too many `ActivityType`s a person usually tracks, this change doesn't improve performance much.

### CODE REVIEW

File: src/main/java/de/dennisguse/opentracks/ui/aggregatedStatistics/AggregatedStatisticsAdapter.java [hunk 1]
```diff
@@ -21,64 +22,43 @@
...
+        if (ActivityType.findByLocalizedString(context, type).isShowSpeedPreferred()) {
             viewHolder.setSpeed(aggregatedStatistic);
         } else {
             viewHolder.setPace(aggregatedStatistic);
         }
-
-        return convertView;
     }
 
-    public AggregatedStatistics swapData(AggregatedStatistics aggregatedStatistics) {
```
Comment (dennisguse) [inline src/main/java/de/dennisguse/opentracks/ui/aggregatedStatistics/AggregatedStatisticsAdapter.java:78]: Nitpicking: diff is easier to read when methods are not switched.
Switching getCategories() and swapData() shows that swapData() actually didn't change.

PS/ Could it be that your AndroidStudio as a cleanup?
Also happened for instance variables (context and aggregatedStatistics).

**Review by dennisguse — APPROVED** [review]
Besides the re-ordering: looks good.

**Comment by dennisguse** [comment]
And you can add `Part of #1613` to the commit message (2nd line).

And that's all the nitpicking :)

File: src/main/java/de/dennisguse/opentracks/ui/aggregatedStatistics/AggregatedStatisticsAdapter.java [hunk 2]
```diff
@@ -183,9 +169,7 @@ private void setCommonValues(AggregatedStatistics.AggregatedStatistic aggregated
 
         private int getIcon(AggregatedStatistics.AggregatedStatistic aggregatedStatistic) {
             String localizedActivityType = aggregatedStatistic.getActivityTypeLocalized();
-            String iconValue = ActivityType.findByLocalizedString(context, localizedActivityType)
-                    .getIconId();
-            return ActivityType.findBy(iconValue)
+            return ActivityType.findByLocalizedString(context, localizedActivityType)
```
Comment (dennisguse) [inline src/main/java/de/dennisguse/opentracks/ui/aggregatedStatistics/AggregatedStatisticsAdapter.java:172]: Nice find :)

**Comment by lmarz** [comment]
I don't really mind the nitpicking. It often improves the code :)

btw. marker list and track list might take a while longer, since there is more stuff happening with them (e.g. selection and cursors).

**Comment by dennisguse** [comment]
Regarding TrackList and MarkerList: this code is ancient (aka still from MyTracks, so likely 10 years old) and was rarely touched. 
IIRC correctly it has the assumption that both share some code - probably not really helpful to keep it this way.

**Pull request merged**